tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ProjectEncodingTest.java

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.tests.resources;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.eclipse.core.internal.resources.PreferenceInitializer;
+import org.eclipse.core.internal.resources.ValidateProjectEncoding;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.hamcrest.*;
+import org.junit.Test;
+
+/**
+ * Test for integration of marker
+ * {@link org.eclipse.core.resources.ResourcesPlugin#PREF_MISSING_ENCODING_MARKER_SEVERITY}.
+ */
+public class ProjectEncodingTest extends ResourceTest {
+
+	private static final int IGNORE = -1;
+
+	private IProject project;
+
+	@Override
+	protected void tearDown() throws Exception {
+		if (project != null) {
+			project.delete(true, null);
+		}
+		InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES).putInt(
```

Comment:
Wasn't this now done with a preference store (in production code) instead of working on the node directly?

Reply:
Maybe I'm missing something here, but preference store is an API of JFace. JFace is a UI framework/library and cannot/should not be used in backend plug-ins like this one.
If I'm understanding it correctly, the preference store used in the UI also uses the this instance scope node:

```
missingEncodingSeverityCombo.setPreferenceStore(new ScopedPreferenceStore(InstanceScope.INSTANCE, ResourcesPlugin.PI_RESOURCES));
```

Reply:
Never mind, I was thinking of ```ResourcesPlugin.getPlugin().getPluginPreferences()```. I checked your code again, you are using the preference service (which is fine) and setting a value with it is not trivial.